transforms/helpers/util/get-telemetry-for.js

```diff
@@ -0,0 +1,54 @@
+const fs = require("fs-extra");
+const path = require("path");
+const walkSync = require("walk-sync");
+const getRepoInfo = require("git-repo-info");
+const Cache = require("sync-disk-cache");
+
+const gitInfo = getRepoInfo();
+const cache = new Cache(`native-class-codemod-${gitInfo.sha}`);
+
+const telemetry = cache.has("telemetry")
+  ? JSON.parse(cache.get("telemetry").value)
+  : {};
+const addonFilePaths = {};
+
+let packagePaths = walkSync("./", {
+  globs: ["**/package.json"],
+  ignore: ["node_modules/**"]
+});
+
+for (let packagePath of packagePaths) {
+  let { name } = fs.readJsonSync(packagePath);
+
+  addonFilePaths[path.dirname(path.resolve(".", packagePath))] = name;
+}
+
+/**
+ * Get the runtime data for the file being transformed
+ *
+ * @param {String} runtimeConfigPath Configuration file path (Absolute)
+ * @param {String} filePath Absolute path of the file to read data from
+ * @returns {Object} Runtime configuration object
+ */
+module.exports = function getTelemetryData(filePath) {
+  let fileSegments = filePath.split("/");
+  let addonSegments = [];
+
+  while (fileSegments.length > 0) {
+    addonSegments.push(fileSegments.shift());
+
+    if (addonFilePaths[addonSegments.join("/")]) {
```

Comment:
Hmm, I don't think this much dynamism is correct. There can be **at most** one `/` in an addon name and only if the first character of the `name` is an `@`.

Reply:
the `addonFilePath` map is from absolute file paths -> addon name. We're receiving an absolute file path here, breaking it into pieces, and then combining each piece until we either find an addon name in the map, or run out of pieces. Since the files could be anywhere on a system, and modules can be nested within an addon, I don't think we can predict where to break the file path we receive.

Reply:
Gotcha! Thank you, I thought this was going to be the in-browser module path and we were mapping it the other way around (returning the on disk path). Sorry about that.